go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -718,6 +720,7 @@ func (npw *nodePortWatcher) DeleteService(service *kapi.Service) error {
 func (npw *nodePortWatcher) SyncServices(services []interface{}) error {
 	var err error
 	var errors []error
+	initCTZones()
```

Comment:
probably would be better to do this in newSharedGateway, since the zones apply to more than just the service watchers. For example, the openflow manager.

Reply:
will try to move to newSharedGateway call, it makes more sense to be there.